code/game/objects/items/reagent_containers/food/snacks.dm

```diff
@@ -2792,3 +2792,76 @@
 		if("spiced apples", "chocolate brownie", "sugar cookie", "choco bar", "crayon")
 			icon_state = "dessert"
 			list_reagents = list(/datum/reagent/consumable/nutriment = 2, /datum/reagent/consumable/sugar = 1)
+
+
+/obj/item/reagent_containers/food/snacks/lollipop
+	name = "lollipop"
+	desc = "A delicious lollipop."
+	icon = 'icons/obj/items/lollipop.dmi'
+	icon_state = "lollipop_stick"
+	item_state = "lollipop_stick"
+	flags_equip_slot = ITEM_SLOT_MASK
+	list_reagents = list(/datum/reagent/consumable/nutriment = 10, /datum/reagent/consumable/sugar = 15)
+	var/mutable_appearance/head
+	var/headcolor = rgb(0, 0, 0)
+	var/succ_dur = 180
+	var/succ_int = 100
+	var/next_succ = 0
+	tastes = list("candy" = 1)
+ 
+/obj/item/reagent_containers/food/snacks/lollipop/Initialize()
+	. = ..()
+	head = mutable_appearance('icons/obj/items/lollipop.dmi', "lollipop_head")
+	change_head_color(rgb(rand(0, 255), rand(0, 255), rand(0, 255)))
+ 
+//makes lollipops actually wearable as masks and still edible the old fashioned way.
+/obj/item/reagent_containers/food/snacks/lollipop/proc/handle_reagents()
+	var/mob/living/carbon/C = loc
+	if(src == C.wear_mask) // if it's in the human/monkey mouth, transfer reagents to the mob
```

Comment:
your already checking this on equip and only processing when its equipped, is this check also needed?